google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedSelectStatementsTests.java

```diff
@@ -80,13 +81,13 @@ public void setup() {
   @Test
   public void selectLockAcquisitionTest() {
     // the translated statement must NOT show locking statements.
-    testStatementTranslation(x -> {
+    List<String> statements = getStatementTranslation(x -> {
       Query q = x.createQuery("select s from SubTestEntity s")
           .setFirstResult(8).setMaxResults(15);
       q.setLockMode(LockModeType.PESSIMISTIC_READ);
       q.list();
-    }, "select subtestent0_.id as id1_1_, subtestent0_.id1 as id2_1_, "
-        + "subtestent0_.id2 as id3_1_ from SubTestEntity subtestent0_ limit ? offset ?");
+    });
+    assertThat(statements.get(0)).doesNotContain("for update");
```

Comment:
Why remove the actual string? having the actual expected string tests for several other things other than just not having "for update" such as  the  offset and limit appearing

Reply:
Yeah I wanted to keep it, but unfortunately Hibernate 6 behaves non-deterministically; i.e. the assertion failed sometimes because half the time it would rearrange the terms in the query in one way, half the time in another.

I changed it here since we don't really care too much about the actual contents of the query but the existence/non-existence certain keywords.

Reply:
This may become a much bigger problem down the road for testing. Can we find out why this is happening, and whether there is a way to disable this non-determinism?

Reply:
The cause appears to be related to the new SQM strategy they implemented for Hibernate 6. I couldn't find a way to disable this non-determinism; the source is likely something fundamental in their algorithm.

The non-determinism is not going to be a significant problem for testing - as long as the query generated is valid it's okay.

The only type of test this precludes us from doing are unit tests in which we input some HQL query and expect some SQL to come out - but moving forward I don't see us adding much more of these kind of tests since we are past the development stage, and we really care about verifying the queries against Real Spanner rather than just comparing the actual sql string against some expected one.